packages/devtools_app/lib/src/debugger/debugger_screen.dart

```diff
@@ -370,3 +370,73 @@ class _DebuggerStatusState extends State<DebuggerStatus> with AutoDisposeMixin {
     return 'paused$reason$fileName $pos';
   }
 }
+
+class FloatingDebuggerControls extends StatelessWidget {
+  @visibleForTesting
+  static const debuggerControlsKey = Key('debugger controls');
+
+  @visibleForTesting
+  static const debuggerControlsResumeKey = Key('debugger controls - resume');
+
+  @visibleForTesting
+  static const debuggerControlsStepKey = Key('debugger controls - step');
+
+  @override
+  Widget build(BuildContext context) {
+    final controller = Provider.of<DebuggerController>(context);
+    return ValueListenableBuilder<bool>(
+      valueListenable: controller.isPaused,
+      builder: (context, paused, _) {
+        if (!paused) return const SizedBox();
```

Comment:
nit: rather than having this become a sized box when not paused consider making the ui unclickable when not paused and providing an animated transition showing and hiding the ui.

Reply:
good idea. done - see pr description for gif